DOCS: improve migration docs

duckets opened the pull request:
### Description
https://jira.unity3d.com/browse/DOCF-5410
Clarify and simplify page so that new system does not look more complex than old system.


### Changes made

Added text & screenshot explaining differences between systems (eg named axes vs actions)
Improved formatting of tables and links.
Move code samples to linked pages.
Group by theme.
Move most common API to top.
Show examples of functions with parameters.
Remove namespaces where not necessary.

### Notes:

Because of the large diff, it may be easier to review the page by viewing the entire formatted page here:
https://github.com/Unity-Technologies/InputSystem/blob/8908df2e3dade953d1b3808eb1a9474d6e2f3a97/Packages/com.unity.inputsystem/Documentation%7E/Migration.md

duckets committed:
migration page improvements WIP

duckets committed:
More WIP for improving Migration page

duckets committed:
more WIP - migrating action-based input

duckets committed:
More WIP on Migration page

duckets committed:
Migration page improvements WIP

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Gamepad.md:
```diff
@@ -198,3 +198,37 @@ The Input System support Switch Pro controllers on desktop computers via the [`S
+
+There are various ways to discover the currently connected devices, as shown in the code samples below.
+
+To query a list of all connected devices (does not allocate; read-only access):
```
This isn't needed for the add/remove onDeviceChange part so could go down with the "To find all gamepads and joysticks" section

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/TableOfContents.md:
```diff
@@ -43,6 +43,6 @@
 * [How do I...?](HowDoI.md)
 * [Architecture](Architecture.md)
-* [Migrating from the old input system](Migration.md)
+* [Migrating from the old Input Manager](Migration.md)
```
This is better phrasing - thanks

lyndon-unity commented on Packages/com.unity.inputsystem/InputSystem/Controls/ButtonControl.cs:
```diff
@@ -87,8 +87,60 @@ public new bool IsValueConsideredPressed(float value)
+        /// Keyboard.current[Key.Space].isPressed
+        ///
+        /// // Using key name.
+        /// (KeyControl)Keyboard.current["space"]).isPressed
```
There is a left bracket missing here.

lyndon-unity replied:
at the start there should be two. e.g.
((KeyControl)Keyboard.current["space"]).isPressed

duckets replied:
with small changes like this where you're sure of the fix, you can can ahead and fix it as a changeset directly in github if you like (saves you even having to check out the branch)

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -3,15 +3,25 @@ uid: input-system-migration
 
+- [Read the introductory documentation first](#read-the-introductory-documentation-first)
+- [Which system is enabled?](#which-system-is-enabled)
+- [List of corresponding API in the old Input Manager new Input System package](#list-of-corresponding-api-in-the-old-input-manager-new-input-system-package)
```
Should probably say
in the old Input Manager vs new Input System package

rather than just
in the old Input Manager new Input System package

lyndon-unity replied:
Seems updated

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -3,15 +3,25 @@ uid: input-system-migration
+- [Which system is enabled?](#which-system-is-enabled)
+- [List of corresponding API in the old Input Manager new Input System package](#list-of-corresponding-api-in-the-old-input-manager-new-input-system-package)
+  - [Action-based input](#action-based-input)
+  - [Directly reading Gamepad and Joystick controls](#directly-reading-gamepad-and-joystick-controls)
```
Why not just have "Gamepad and Joystick" ( as you have Keyboard or Mouse below)

duckets replied:
because it's specifically to do with the "direct" workflow - i.e. not using actions and bindings

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 All of the new APIs listed below are in the `UnityEngine.InputSystem` namespace. The namespace is omitted here for brevity. `UnityEngine.InputSystem` is referenced in full for easy disambiguation.
```
This was already present but slightly confused why it says 

The namespace is omitted here for brevity

but then says 
`UnityEngine.InputSystem` is referenced in full for easy disambiguation.

I feel we could drop the last line
`UnityEngine.InputSystem` is referenced in full for easy disambiguation.

duckets replied:
yes - I'm not sure where that last line came from actually, maybe left over from a previous version of the page.

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
-|--|--|
-[`Input.GetKey`](https://docs.unity3d.com/ScriptReference/Input.GetKey.html)|[`ButtonControl.isPressed`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_isPressed) on the corresponding key:
+- In the old Input Manager, these are defined in the **Axes** list, in the **Input Manager** section of the **Project Settings** window. _(Below, left)_
+- In the new Input System, these are defined in the [Actions Editor](ActionsEditor.md), which can be found in the **Input System Package** section of the **Project Settings** window, or by opening an [Action Asset](ActionAssets.md). _(Below, right)_
```
Love this bit

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 
-// Using Key enum.
-Keyboard.current[Key.Space].isPressed
+__Note:__ In some cases for named axes and buttons, the new Input System requires slightly more code than the old Input Manager, but this results in better performance. This is because in the new Input System, the logic is separated into two parts: the first is to find and store a reference to the action (usually done once, for example in your `Start` method), and the second is to read the action ...
```
Also great

ekcoh replied:
Agree, this provides motivation what is coming - the WHY

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 
-// Using key name.
-((KeyControl)Keyboard.current["space"]).isPressed
+To find and store references to actions, which can be axes or buttons use [`FindAction`](../api/UnityEngine.InputSystem.InputActionAsset.html#UnityEngine_InputSystem_InputActionAsset_FindAction_System_String_System_Boolean_). For example:
```
Might be worth saying 'via the globally available actions set in the projects settings' (and not that you can also reference an individual asset by name if needed and link out to another page for that part)

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 |Input Manager (Old)|Input System (New)|
 |--|--|
-[`Input.GetKeyDown`](https://docs.unity3d.com/ScriptReference/Input.GetKeyDown.html)|Use [`ButtonControl.wasPressedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasPressedThisFrame) on the corresponding key:
+[`Input.GetAxis`](https://docs.unity3d.com/ScriptReference/Input.GetAxis.html)<br/>In the old Input Manager System, all axes are 1D and return float values. For example, to read the horizontal and vertical axes:<br/>`float h = Input.GetAxis("Horizontal");`<br/>`float v = Input.GetAxis("Vertical");`<br/><br/> | Use [`ReadValue`](../api/UnityEngine.InputSystem.InputBindingComposite-1.html#UnityEngi ...
```
Love this showing the new is simpler for 2D

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.GetButton`](https://docs.unity3d.com/ScriptReference/Input.GetButton.html)<br/>Example:<br/>`bool jumpValue = Input.GetButton("Jump");`<br/><br/>|Use [`IsPressed`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_IsPressed_) on the reference to the action to read the button value.<br/>Example:<br/>`bool jumpValue = jumpAction.IsPressed()`.<br/><br/>
+[`Input.GetButtonDown`](https://docs.unity3d.com/ScriptReference/Input.GetButtonDown.html)<br/>Example: `bool jump = Input.GetButtonDown("Jump");`<br/><br/>|Use [`WasPressedThisFrame`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_WasPressedThisFrame_) on the reference to the action to read if the button was pressed this frame.<br/>Example: `bool jumpValue =  ...
+[`Input.GetButtonUp`](https://docs.unity3d.com/ScriptReference/Input.GetButtonUp.html)<br/>Example: `bool jump = Input.GetButtonUp("Jump");`<br/><br/>|Use [`WasReleasedThisFrame`](../api/UnityEngine.InputSystem.InputAction.html#UnityEngine_InputSystem_InputAction_WasReleasedThisFrame_) on the reference to the action to read whether the button was released this frame.<br/>Example: `bool jumpValue  ...
+[`Input.GetAxisRaw`](https://docs.unity3d.com/ScriptReference/Input.GetAxisRaw.html)|Not directly applicable. You can use [`InputControl<>.ReadUnprocessedValue()`](../api/UnityEngine.InputSystem.InputControl-1.html#UnityEngine_InputSystem_InputControl_1_ReadUnprocessedValue) to read unprocessed values from any control.
```
You should probably show a full example here. E.g.

Vector2 moveVector = moveAction.ReadUnprocessedValue<Vector2>().

ekcoh replied:
Might it be relevant to point out that this is equivalent to ReadValue in case the binding do not have any associated processors?

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
````diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
-```
+### Directly reading Gamepad and Joystick controls
+
+Directly reading hardware controls bypasses the new Input System's action-based workflow, which has some benefits and some drawbacks. ([Read more about directly reading devices](./Workflow-Direct.md))
````
I'd suggest we also make a recommendation inline here . E.g. something like

To allow separate the functionality and the control surface we recommend using the actions concept rather than direct access to devices. See [actions-based input] link

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.GetKeyDown`](https://docs.unity3d.com/ScriptReference/Input.GetKeyDown.html)<br/>Example: `Input.GetKeyDown(KeyCode.Space)`<br/><br/>|Use [`wasPressedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasPressedThisFrame) on the corresponding key.<br/> Example: `InputSystem.Keyboard.current.spaceKey.wasPressedThisFrame`<br ...
+[`Input.GetKeyUp`](https://docs.unity3d.com/ScriptReference/Input.GetKeyUp.html)<br/>Example: `Input.GetKeyUp(KeyCode.Space)`<br/><br/>|Use [`wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding key.<br/> Example: `InputSystem.Keyboard.current.spaceKey.wasReleasedThisFrame`<br/>< ...
+[`Input.anyKey`](https://docs.unity3d.com/ScriptReference/Input-anyKey.html)|Use [`onAnyButtonPress`](../api/UnityEngine.InputSystem.InputSystem.html#UnityEngine_InputSystem_InputSystem_onAnyButtonPress).<br/>This also includes controller buttons as well as keyboard keys.
+[`Input.anyKeyDown`](https://docs.unity3d.com/ScriptReference/Input-anyKeyDown.html)|Use [`Keyboard.current.anyKey.wasUpdatedThisFrame`](../api/UnityEngine.InputSystem.Keyboard.html)
```
Not sure the second link works to get to the anyKey inside the keyboard page
Did you plan for it to go here:
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.8/api/UnityEngine.InputSystem.Keyboard.html#UnityEngine_InputSystem_Keyboard_anyKey

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.GetMouseButton`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButton.html)<br/>Example: `Input.GetMouseButton(0)`|Use [`isPressed`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_isPressed) on the corresponding mouse button.<br/>Example: `InputSystem.Mouse.current.leftButton.isPressed`
+[`Input.GetMouseButtonDown`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonDown.html)<br/>Example: `Input.GetMouseButtonDown(0)`|Use [`wasPressedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasPressedThisFrame) on the corresponding mouse button.<br/>Example: `InputSystem.Mouse.current.leftButton.wasPressedThis ...
+[`Input.GetMouseButtonUp`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonUp.html)<br/>Example: `Input.GetMouseButtonUp(0)`|Use [`wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding mouse button.<br/>Example: `InputSystem.Mouse.current.leftButton.wasReleasedThisFra ...
+[`Input.mousePosition`](https://docs.unity3d.com/ScriptReference/Input-mousePosition.html)|Use [`Mouse.current.position.ReadValue()`](../api/UnityEngine.InputSystem.Mouse.html)<br/>__Note__: Mouse simulation from touch isn't implemented yet.
```
For a moment I thought the mouse simulation part might be out of date

But I've only sound the revised - "simulating touch from mouse" (rather than mouse from touch)
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.8/manual/Touch.html#touch-simulation

^ so this comment is just a pointless note ;)

lyndon-unity replied:
I'd suggest putting a code section with this bit
position = Mouse.current.position.ReadValue();

And keeping the Mouse documentation page link separate.
As I think its more clear what to cut and paste into code.

ekcoh replied:
May this be relevant here? https://docs.unity3d.com/Packages/com.unity.inputsystem@1.8/api/UnityEngine.InputSystem.UI.VirtualMouseInput.html

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.GetMouseButtonDown`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonDown.html)<br/>Example: `Input.GetMouseButtonDown(0)`|Use [`wasPressedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasPressedThisFrame) on the corresponding mouse button.<br/>Example: `InputSystem.Mouse.current.leftButton.wasPressedThis ...
+[`Input.GetMouseButtonUp`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonUp.html)<br/>Example: `Input.GetMouseButtonUp(0)`|Use [`wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding mouse button.<br/>Example: `InputSystem.Mouse.current.leftButton.wasReleasedThisFra ...
+[`Input.mousePosition`](https://docs.unity3d.com/ScriptReference/Input-mousePosition.html)|Use [`Mouse.current.position.ReadValue()`](../api/UnityEngine.InputSystem.Mouse.html)<br/>__Note__: Mouse simulation from touch isn't implemented yet.
+[`Input.mousePresent`](https://docs.unity3d.com/ScriptReference/Input-mousePresent.html)|Use [`Mouse.current != null`](../api/UnityEngine.InputSystem.Mouse.html#UnityEngine_InputSystem_Mouse_current).
```
I'd suggest putting a code section with this bit
Mouse.current != null

And keeping the Mouse.Current link separate.
As I think its more clear what to cut and paste into code.

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 |Input Manager (Old)|Input System (New)|
 |--|--|
-[`Input.GetMouseButtonUp`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonUp.html)|Use [`ButtonControl.wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding mouse button:
+[`Input.GetTouch`](https://docs.unity3d.com/ScriptReference/Input.GetTouch.html)|Use [`EnhancedTouch.Touch.activeTouches[i]`](../api/UnityEngine.InputSystem.EnhancedTouch.Touch.html#UnityEngine_InputSystem_EnhancedTouch_Touch_activeTouches)<br/>__Note__: Enable enhanced touch support first by calling [`EnhancedTouch.Enable()`](../api/UnityEngine.InputSystem.EnhancedTouch.EnhancedTouchSupport.html ...
```
Given this is more common input for mobile I think its worth an example.

touch = Input.GetTouch(0);
touch.position

vs

EnhancedTouch.Touch.activeTouches[0].position


TBH the new input system touch page itself would benefit from giving a more explicit example (as the table covers various options but doesn't make it super clear what to type to just get the position or state)
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.8/manual/Touch.html

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.touches`](https://docs.unity3d.com/ScriptReference/Input-touches.html)|[`EnhancedTouch.Touch.activeTouches`](../api/UnityEngine.InputSystem.EnhancedTouch.Touch.html#UnityEngine_InputSystem_EnhancedTouch_Touch_activeTouches)<br/>__Note__: Enable enhanced touch support first by calling [`EnhancedTouch.Enable()`](../api/UnityEngine.InputSystem.EnhancedTouch.EnhancedTouchSupport.html#UnityEng ...
+[`Input.touchPressureSupported`](https://docs.unity3d.com/ScriptReference/Input-touchPressureSupported.html)|No corresponding API yet.
+[`Input.touchSupported`](https://docs.unity3d.com/ScriptReference/Input-touchSupported.html)|[`Touchscreen.current != null`](../api/UnityEngine.InputSystem.Touchscreen.html#UnityEngine_InputSystem_Touchscreen_current)
+[`UnityEngine.TouchScreenKeyboard`](https://docs.unity3d.com/ScriptReference/TouchScreenKeyboard.html)|No corresponding API yet. Use `TouchScreenKeyboard` for now.
```
Maybe here we should be saying - this is still present as its not part of the old Input Manager API

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+[`Input.compass`](https://docs.unity3d.com/ScriptReference/Input-compass.html)|No corresponding API yet.
+[`Input.compensateSensors`](https://docs.unity3d.com/ScriptReference/Input-compensateSensors.html)|[`InputSettings.compensateForScreenOrientation`](../api/UnityEngine.InputSystem.InputSettings.html#UnityEngine_InputSystem_InputSettings_compensateForScreenOrientation).
+[`Input.deviceOrientation`](https://docs.unity3d.com/ScriptReference/Input-deviceOrientation.html)|No corresponding API yet.
+[`Input.gyro`](https://docs.unity3d.com/ScriptReference/Input-gyro.html)|The `UnityEngine.Gyroscope` class is replaced by multiple separate sensor Devices in the new Input System:<br/>[`Gyroscope`](../api/UnityEngine.InputSystem.Gyroscope.html) to measure angular velocity.<br/>[`GravitySensor`](../api/UnityEngine.InputSystem.GravitySensor.html) to measure the direction of gravity.<br/>[`AttitudeS ...
```
I see you have specific examples below the main section. I was wondering if you needed this line. Perhaps splitting the header links into the next few sections.

Happy to leave this bit though as its minor.

lyndon-unity reviewed (approved):
This update is a great improvement.
I've made some suggestions but even in this form its a step forward so approving anyway.

I've love to see the updates if you agree, but can somewhat pre approve.

ekcoh commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
 |Input Manager (Old)|Input System (New)|
 |--|--|
-[`Input.GetMouseButtonUp`](https://docs.unity3d.com/ScriptReference/Input.GetMouseButtonUp.html)|Use [`ButtonControl.wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding mouse button:
+[`Input.GetTouch`](https://docs.unity3d.com/ScriptReference/Input.GetTouch.html)|Use [`EnhancedTouch.Touch.activeTouches[i]`](../api/UnityEngine.InputSystem.EnhancedTouch.Touch.html#UnityEngine_InputSystem_EnhancedTouch_Touch_activeTouches)<br/>__Note__: Enable enhanced touch support first by calling [`EnhancedTouch.Enable()`](../api/UnityEngine.InputSystem.EnhancedTouch.EnhancedTouchSupport.html ...
```
Why isn't the old input API also show index variable if new is?

lyndon-unity replied:
Looks like this is updated now.

ekcoh reviewed (approved):
Looks great @duckets, a big improvement to the migration docs. Added some minor comments.

duckets committed:
Amendments from PR review comments

lyndon-unity commented on Packages/com.unity.inputsystem/Documentation~/Migration.md:
```diff
@@ -29,207 +39,113 @@ There are scripting symbols defined which allow you to use conditional compilati
 |Input Manager (Old)|Input System (New)|
 |--|--|
-[`Input.GetKeyUp`](https://docs.unity3d.com/ScriptReference/Input.GetKeyUp.html)|Use [`ButtonControl.wasReleasedThisFrame`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_wasReleasedThisFrame) on the corresponding key:
+[`Input.GetButton`](https://docs.unity3d.com/ScriptReference/Input.GetButton.html)<br/>Example: `Input.GetKey(KeyCode.JoystickButton0)`<br/><br/>|Use [`isPressed`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_isPressed) on the corresponding Gamepad button.<br/>Example: `InputSystem.GamePad.current.buttonNorth.isPressed`.<br/>
```
The example given is GeyKey but this is a GetButton row so it looks like the example is wrong

lyndon-unity replied:
I've fixed this to refer to GetKey as we are talking about direct input here.

ekcoh committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

ekcoh committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

ekcoh committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

lyndon-unity committed:
Update Migration.md

Corrected GetKey/GetButton mismatch in the joysticks section

(GetButton is analogous to 'actions' and GetKey is analogous to direct device access)

ekcoh committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

lyndon-unity commented:
We seem to lack migration info for GetPenEvent, GetLastPenContactEvent, ResetPenEvents, ClearLastPenContactEvent

All could refer to Pen.current

lyndon-unity commented:
I also think imeCompositionMode should refer to Keyboard.current.SetIMEEnabled(true);

Currently Input.imeIsSelected refers to both Keyboard.current.imeSelected and Keyboard.current.SetIMEEnabled(true);
I think this should only refer to Keyboard.current.imeSelected

lyndon-unity commented:
Some issues remain with KeyControl.displayName being referenced - This field is in InputControl (which KeyControl inherits from)

isPressed in Packages/com.unity.inputsystem/InputSystem/Controls/ButtonControl.cs Line 147wasPressedThisFrame in Packages/com.unity.inputsystem/InputSystem/Controls/ButtonControl.cs Line 185
wasReleasedThisFrame in Packages/com.unity.inputsystem/InputSystem/Controls/ButtonControl.cs Line 21

duckets committed:
Fix broken api links for KeyControl.displayName

duckets committed:
Formatting fix

lyndon-unity reviewed (approved):
Updates look good

duckets committed:
Updates to migration guide for Pen events

duckets committed:
Updates to migration guide for Keyboard IME text composition

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
Removed migration info for Input.mousePresent

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

lyndon-unity committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
Formatting fix

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

AlexTyrer committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs

duckets committed:
fix xml for ButtonControl

duckets committed:
xml fix

duckets committed:
xml doc fix

duckets committed:
whitespace format fix

duckets committed:
added para tags for example plain text

lyndon-unity committed:
Merge branch 'develop' into DOCF-5410-improve-migration-docs
